### PR TITLE
Migrate some expressions away from serokell-profiles

### PR DESCRIPTION
**Pull request by notgne2**
I also fixed flake-compat to work with the newer lockfile

`common.nix` is from `profiles/servers/default.nix` in `serokell-profiles`, `nixosModules.defaults` in `flake.nix` is from `modules/default.nix` in `serokell-profiles`. The diff is huge, but it's mostly just copying files.

### CODE REVIEW

**Review by balsoft — COMMENTED** [review]
`common.nix` doesn't feel right at the root of the repo, maybe `modules` ?

**Comment by zhenyavinogradov** [comment]
Could you give a tldr on why we want to migrate the modules? I saw some discussions on slack, but they are hard to follow sometimes

**Comment by notgne2** [comment]
"Modernizing" serokell-profiles is a task that is encountering more an more issues, so the only reasonable solution is to abandon it for new deployment, but we still need to re-use modules that it currently contains (especially as we are trying to deconstruct and decommission Jupiter). This repository is a nice standard place where both serokell-profiles and our newer deployments can pull from.

File: overlay/default.nix [hunk 1]
```diff
@@ -44,4 +59,4 @@ final: prev: {
         '';
     };
   };
-}
+})
```
Comment (mkaito) [inline overlay/default.nix:62]: WTB newline

File: overlay/youtrack_rev.json.license [hunk 2]
```diff
@@ -0,0 +1,3 @@
+SPDX-FileCopyrightText: 2020 Serokell <https://serokell.io/>
+
+SPDX-License-Identifier: MPL-2.0
```
Comment (mkaito) [inline overlay/youtrack_rev.json.license:3]: Newline

**Review by mkaito — CHANGES_REQUESTED** [review]
Do we still actually want an overlay, or can we just use flake inputs?

File: overlay/youtrack_rev.json.license (hunk 2 above)
Comment (notgne2) [reply overlay/youtrack_rev.json.license:3]: This is automatically generated by `reuse`, so manually adding a newline seems a little unusual

Comment (mkaito) [reply overlay/youtrack_rev.json.license:3]: Oh, ok, I'll stop complaining about `.license` files.

File: modules/common.nix [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  services.nginx = {
+    appendHttpConfig = ''
+      if_modified_since off;
+      proxy_set_header        Host $host;
+      proxy_set_header        X-Real-IP $remote_addr;
+      proxy_set_header        X-Forwarded-For $proxy_add_x_forwarded_for;
+      proxy_set_header        X-Forwarded-Proto $scheme;
+      proxy_set_header        X-Forwarded-Host $host;
+      proxy_set_header        X-Forwarded-Server $host;
+    '';
```
[the remark this answers is not in the document]
Comment (notgne2) [reply modules/common.nix:56]: Looks like the lines from it (besides the headers which we have an exact copy of) are
```nginx
proxy_redirect          off;
proxy_connect_timeout   90;
proxy_send_timeout      90;
proxy_read_timeout      90;
proxy_http_version      1.0;
```
are any of those options problematic?

File: modules/common.nix [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+  services.nginx.package = pkgs.nginxStable.override {
+    modules = with pkgs.nginxModules; [ brotli ];
+  };
+
+  users.mutableUsers = false;
+
+  nix.nixPath = [
+    "nixpkgs=${pkgs.path}:nixos-config=/etc/nixos/configuration.nix:/nix/var/nix/profiles/per-user/root/channels"
+  ];
```
[the remark this answers is not in the document]
Comment (notgne2) [reply modules/common.nix:78]: @balsoft should know better, but I think `nixpkgs=${pkgs.path}` is good and will do what it should, `nixos-config=/etc/nixos/configuration.nix` is useless, and `/nix/var/nix/profiles/per-user/root/channels` I don't understand the purpose of

File: modules/common.nix [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    automatic = true;
+    # delete so there is 15GB free, and delete very old generations
+    # delete-older-than by itself will still delete all non-referenced packages (ie build dependencies)
+    options = lib.mkForce ''
+      --max-freed "$((15 * 1024**3 - 1024 * $(df -P -k /nix/store | tail -n 1 | ${pkgs.gawk}/bin/awk '{ print $4 }')))" --delete-older-than 14d'';
+  };
+
+  programs.mosh.enable = true;
+
+  # TODO: setup Bastion server
```
[the remark this answers is not in the document]
Comment (notgne2) [reply modules/common.nix:44]: Is this still planned in any way?

File: flake.nix [hunk 6]
```diff
@@ -7,23 +7,52 @@
...
+        services.youtrack.virtualHost = lib.mkOptionDefault "youtrack";
+      };
+
       acme-sh = import ./modules/acme-sh.nix;
       vault-secrets = import ./modules/vault-secrets.nix;
+
+      serokell-users = import ./modules/serokell-users.nix;
+      hackage-search = import ./modules/services/hackage-search.nix;
+      hermetic = import ./modules/services/hermetic.nix;
+      matomo = import ./modules/services/matomo.nix;
```
[the remark this answers is not in the document]
Comment (notgne2) [reply flake.nix:48]: Is it planned to be used for anything? There's no point removing it if we're going to add it back

File: modules/common.nix (hunk 4 above)
[the remark this answers is not in the document]
Comment (balsoft) [reply modules/common.nix:78]: `nixpkgs=${pkgs.path}` is alright (although I prefer `nixpkgs=/etc/nixpkgs` and then `environment.etc.nixpkgs.source = pkgs.path;` because it changes right after activation instead of having to relogin to get new nixpkgs), everything else should go

[the remark this answers is not in the document]
Comment (notgne2) [reply modules/common.nix]: I've updated this now in the last commit to work the way you suggested, I also used a similar method for overlays but I'm not entirely certain yet if this even works (Nix might not resolve the paths within the overlay file correctly this way)

File: modules/common.nix (hunk 3 above)
[the remark this answers is not in the document]
Comment (mkaito) [reply modules/common.nix:56]: Not that I'm aware of.

File: modules/common.nix (hunk 5 above)
[the remark this answers is not in the document]
Comment (mkaito) [reply modules/common.nix:44]: No, this was a very early idea for infrastructure design which we never actually did. Up to you now if you would like to follow up on it. This comment is irrelevant either way.

File: flake.nix (hunk 6 above)
[the remark this answers is not in the document]
Comment (mkaito) [reply flake.nix:48]: No, we use Prometheus.

[the remark this answers is not in the document]
Comment (notgne2) [reply flake.nix:48]: Matomo is a web analytics platform (formerly Piwik), I've not heard of Prometheus being used in that way

File: modules/common.nix (hunk 3 above)
[the remark this answers is not in the document]
Comment (notgne2) [reply modules/common.nix]: I've re-enabled `recommendedProxySettings` now and removed the manual `proxy_set_header` directives

**Comment by balsoft** [comment]
Doesn't actually have `ssh-keys.nix`

**Review by mkaito — CHANGES_REQUESTED** [review]
We can merge once we have CI checks in place

**Comment by mkaito** [comment]
Okay, now go actually set up a pipeline in buildkite.

**Comment by notgne2** [comment]
@mkaito I already did

**Comment by mkaito** [comment]
> Fix CI error

:coolstory:

**Review by mkaito — APPROVED** [review]
With CI fixed, we can merge this now.
